src/charm.py

```diff
@@ -261,6 +261,9 @@ def get_secret(self, scope: Scopes, key: str) -> Optional[str]:
             raise RuntimeError("Unknown secret scope.")
 
         peers = self.model.get_relation(PEER)
+        if not peers:
```

Comment:
Won't this charm need `DataPeerData` and `DataPeerUnitData`, like we see when the lib was updated in the other charm repos?

Reply:
oh yes, in fact this PR is actually still a WIP, I forgot to change it to draft! Sorry about that

Reply:
PR is improved now, feel free to re-review!